### DISCUSSION TITLE
Curso de pós graduação em Desenvolvimento Front-end

### DISCUSSION THREAD

**Discussion by vmagalhaes** [frontendbr/forum#1606] [Discussão]
Opa Fronts,

Me formei em Sistemas, e já queria engatar uma pós,
Tenho como foco o front-end, trabalho à 3 anos diretamente com Angular e quero cada vez mais me especializar, a minha dúvida seria:

Esse curso da IGTI, alguém já ouviu falar? Sabem se é bom?
https://www.igti.com.br/cursos/mba-em-desenvolvimento-front-end/

E outra, certificados, gostaria de tirar alguns, saberiam me indicar uns certificados legais, que valem algo mesmo? Queria fazer esse investimento para a minha carreira.

Valeu!

**Comment by caioincau** [comment]
Cara, minha opinião sobre pós ficada em tecnologia é que isto é uma perda de investimento, tem meios bem mais interessantes de se aprender do que em uma pós, só faria uma pós se fosse um tema mais abrangente, tipo gestão de pessoas, administração ou algo que me agregasse extra código e fosse menos sensível ao tempo.

Sobre certificados pra front end eu desconheço tb, geralmente isso é coisa de empresa grande tipo Oracle, Microsoft e afins.

**Reply by devleonardo** [reply]
Tem certificações da w3schools. Alguns concursos até aceitam como prova de títulos. Pelo menos foi o que vi na inscrição da FAB, no ultimo concurso para o quadro de sargento temporário.

**Comment by vmagalhaes** [comment]
> Cara, minha opinião sobre pós ficada em tecnologia é que isto é uma perda de investimento, tem meios bem mais interessantes de se aprender do que em uma pós, só faria uma pós se fosse um tema mais abrangente, tipo gestão de pessoas, administração ou algo que me agregasse extra código e fosse menos sensível ao tempo.
> 
> Sobre certificados pra front end eu desconheço tb, geralmente isso é coisa de empresa grande tipo Oracle, Microsoft e afins.

Bom ponto, o problema desse curso que parece ser sensível ao tempo mesmo. Eu estou procurando algo que além de agregar conhecimento sobre o desenvolvimento frontend, impulsione o meu currículo. Além da experiência.

Uma coisa boa seria certificados.

**Comment by ViniciusGularte** [comment]
Sigo na mesma opinião do @caioincau , eu sai da graduação e engatei direto numa pós na tecnologia, e foi perda de tempo e dinheiro,  poderia ter pagado algum curso/bootcamp foda... eu pretendo muito iniciar outra faculdade na área de finanças/humanas, conhecimento nunca é demais, também desconheço certificações no ramo frontend, só sei das certificações do nodejs ( 300 dolares  💀 )

**Comment by caioincau** [comment]
Hoje em dia da para fazer Harvard business a distância e se duvidar sai mais barato que mta faculdade BR

**Comment by onlurking** [comment]
Cara, depende muito do teu foco, se é se especializar em HTML, Acessibilidade, CSS ou Javascript(vanilla ou em algum framework em específico por exemplo), tem que escolher um e se aprofundar no melhores materiais disponíveis sobre o assunto, é impossível esperar que uma pós ou bootcamp cubra esses assuntos decentemente.

De CSS por exemplo tem o [The CSS Layout Workshop](https://thecssworkshop.com/toc) da [Rachel Andrew](https://rachelandrew.co.uk/css/) que é uma das mentes por trás do CSS Grid Layout e ela está no CSS Working Group, que é o comitê responsável por criar as especificações do CSS. 

Tem o [The Responsive Web Design Bootcamp](https://scrimba.com/g/gresponsive) do [Kevin Powell](https://www.youtube.com/channel/UCJZv4d5rbIKd4QHMPkcABCw), que é uma referência de CSS também, recomendo dar uma olhada no canal dele no youtube.

De referência nacional em CSS os cursos do [André Rafael da Origamid](https://www.origamid.com/) são incríveis, tenho o certificado de Flexbox e Grid Layout dele, e só pra tu ter uma ideia, botou o nanodegree de front-end que fiz da Udacity(material feito em parceria pela Google) no chinelo, se quiser ter uma noção da qualidade dá uma olhada nos guias de referência dele em [flexbox](https://origamid.com/projetos/flexbox-guia-completo/) e [grid layout](https://www.origamid.com/projetos/css-grid-layout-guia-completo/).

De Javascript tem os materiais do [Eric Elliot](https://ericelliottjs.com), que é o cara por trás de vários artigos incríveis de Javascript como o [The Two Pillars of JavaScript](https://medium.com/javascript-scene/the-two-pillars-of-javascript-ee6f3281e7f3), tem os livros da série [You Don't Know Javascript](https://github.com/getify/You-Dont-Know-JS#titles) do Kyle Simpson e também os cursos dele no [Frontend Masters](https://frontendmasters.com/teachers/kyle-simpson/), conteúdo nacional tem os cursos do [Daciuk](https://queroser.ninja/promocoes/) e do [Willian Justen](https://www.udemy.com/user/willian-justen-de-vasconcellos/) que são duas referências em frontend.

De React tem os cursos do [Tyler Mcginnis](https://tylermcginnis.com/courses/) que está por trás da emenda do [nanodegree de React](https://www.udacity.com/course/react-nanodegree--nd019) da Udacity, de conteúdo nacional de React tem o curso [React Ninja](https://www.udemy.com/course/curso-reactjs-ninja/) do Fernando Daciuk, os cursos do [devPleno](https://www.devpleno.com) do Tulio Faria, tem o material da [Rocketseat](https://rocketseat.com.br)(estou fazendo o bootcamp deles atualmente, é bem legal mas peca bastante em boas práticas).

Certificados oficiais em frontend não existem, quer dizer, tem os certificados da [W3C pela edX](https://www.edx.org/professional-certificate/front-end-web-developer-9) mas os cursos são bem básicos, não valem a pena, e tem os da [W3Schools](https://www.w3schools.com/cert/) mas eles tem a imagem um pouco suja por ter um histórico de [ensinar coisas erradas](https://web.archive.org/web/20120502220433/http://w3fools.com/).

De qualquer maneira, portfolio é muito mais relevante que qualquer certificado. Vai lhe agregar muito mais você escolher um curso/bootcamp que é recheado de projetos, fazer eles, colocar no Github, e melhorar eles após a finalização do curso(pesquisar sobre boas práticas, documentação e refatorar, procurar material fora do curso, na minha opinião o aprendizado real acontece aqui).

**Reply by AdmoSouza** [reply]
Caraca o cara é brabo ... valeu pelas dicas mano

**Reply by emersonjds** [reply]
mandou benzaço !

**Reply by ltoniati** [reply]
Vou até salvar isso kkkkk

**Reply by tcharlesfr** [reply]
Isso não é um usuário comum, é um prefeito! (salvei a resposta aqui)

**Comment by kivervinicius** [comment]
@vmagalhaes Opa, cara, eu tenho 2 pós na área de tecnologia e uma na área de docente, fui professor em faculdades, foi um dos motivos de ter feito elas, porém para empresas comuns pós não agrega não, o cara só vai olhar e falar (olha que legal ele estudou em tal faculdade e tal curso) e vai perguntar se você tem experiência em alguma tecnologia especifica, meu conselho, faça um nanodegree, hoje muitas empresas enxergam isso com ótimos olhos por ser algo bem criterioso e focado, pós só se quiser fazer bonito com a familia ou um dia pensar em ministrar aulas/cursos que é o meu caso.

**Comment by ebraimcarvalho** [comment]
> Cara, depende muito do teu foco, se é se especializar em HTML, Acessibilidade, CSS ou Javascript(vanilla ou em algum framework em específico por exemplo), tem que escolher um e se aprofundar no melhores materiais disponíveis sobre o assunto, é impossível esperar que uma pós ou bootcamp cubra esses assuntos decentemente.
> 
> De CSS por exemplo tem o [The CSS Layout Workshop](https://thecssworkshop.com/toc) da [Rachel Andrew](https://rachelandrew.co.uk/css/) que é uma das mentes por trás do CSS Grid Layout e ela está no CSS Working Group, que é o comitê responsável por criar as especificações do CSS.
> 
> Tem o [The Responsive Web Design Bootcamp](https://scrimba.com/g/gresponsive) do [Kevin Powell](https://www.youtube.com/channel/UCJZv4d5rbIKd4QHMPkcABCw), que é uma referência de CSS também, recomendo dar uma olhada no canal dele no youtube.
> 
> De referência nacional em CSS os cursos do [André Rafael da Origamid](https://www.origamid.com/) são incríveis, tenho o certificado de Flexbox e Grid Layout dele, e só pra tu ter uma ideia, botou o nanodegree de front-end que fiz da Udacity(material feito em parceria pela Google) no chinelo, se quiser ter uma noção da qualidade dá uma olhada nos guias de referência dele em [flexbox](https://origamid.com/projetos/flexbox-guia-completo/) e [grid layout](https://www.origamid.com/projetos/css-grid-layout-guia-completo/).
> 
> De Javascript tem os materiais do [Eric Elliot](https://ericelliottjs.com), que é o cara por trás de vários artigos incríveis de Javascript como o [The Two Pillars of JavaScript](https://medium.com/javascript-scene/the-two-pillars-of-javascript-ee6f3281e7f3), tem os livros da série [You Don't Know Javascript](https://github.com/getify/You-Dont-Know-JS#titles) do Kyle Simpson e também os cursos dele no [Frontend Masters](https://frontendmasters.com/teachers/kyle-simpson/), conteúdo nacional tem os cursos do [Daciuk](https://queroser.ninja/promocoes/) e do [Willian Justen](https://www.udemy.com/user/willian-justen-de-vasconcellos/) que são duas referências em frontend.
> 
> De React tem os cursos do [Tyler Mcginnis](https://tylermcginnis.com/courses/) que está por trás da emenda do [nanodegree de React](https://www.udacity.com/course/react-nanodegree--nd019) da Udacity, de conteúdo nacional de React tem o curso [React Ninja](https://www.udemy.com/course/curso-reactjs-ninja/) do Fernando Daciuk, os cursos do [devPleno](https://www.devpleno.com) do Tulio Faria, tem o material da [Rocketseat](https://rocketseat.com.br)(estou fazendo o bootcamp deles atualmente, é bem legal mas peca bastante em boas práticas).
> 
> Certificados oficiais em frontend não existem, quer dizer, tem os certificados da [W3C pela edX](https://www.edx.org/professional-certificate/front-end-web-developer-9) mas os cursos são bem básicos, não valem a pena, e tem os da [W3Schools](https://www.w3schools.com/cert/) mas eles tem a imagem um pouco suja por ter um histórico de [ensinar coisas erradas](https://web.archive.org/web/20120502220433/http://w3fools.com/).
> 
> De qualquer maneira, portfolio é muito mais relevante que qualquer certificado. Vai lhe agregar muito mais você escolher um curso/bootcamp que é recheado de projetos, fazer eles, colocar no Github, e melhorar eles após a finalização do curso(pesquisar sobre boas práticas, documentação e refatorar, procurar material fora do curso, na minha opinião o aprendizado real acontece aqui).

O ruim desse comentário é que ele acaba e não tem como dar uma estrela pra ficar revisitando de tempos em tempos! Mandei pro email pra salvar essa pérola! Valeu mano!!!

**Comment by caiqalmeida** [comment]
@onlurking caraca, que comentário completaço 
Obrigado mano !

**Comment by cristofersousa** [comment]
No inicio do ano eu dei uma baita pesquisada sobre Pós Graduação voltado para Front-end.

E também encontrei essas:

- [FIAP](https://www.fiap.com.br/mba/mba-em-full-stack-developer-microservices-cloud-e-iot/)
- [IGTI](https://www.igti.com.br/cursos/mba-em-desenvolvimento-front-end/)

Fiz um bate papo com um professor lá do curso e ele me disse que era bem para quem estava trocando de área e queria conhecer sobre Front-end, não é o seu cenário que já atua há alguns anos nesse campo. 

Sobre a abordagem do CaioIncau sobre custos x  vantagens, realmente fazer uma pós graduação em front-end é bem complicado pois os professores que estão lá atuando não estão no mercado, então a grade é defasada e o conhecimento deles da qual tu busca também não será algo tão eficaz. 

Eu fiz uma pós graduação na [UFSCar](http://latosensu.dc.ufscar.br/desenvolvimento-de-software-para-web-2019) com foco em Desenvolvimento de Software para Web a grade e os professores eram bem defasados, mas depois de algumas considerações que fiz para o coordenador do curso, ele fez uma baita "refatorada" colocando React, Node entre outras tecnologias do momento na grade.  

Conforme o Kiver pontuou para quem procura pós graduação a vantagem é com título tu poderá atuar para dar aula em faculdade mas para o mercado de trabalho não pesa tanto, para mim foi válido pois gosto dessa viés de aula.

Conversando com o tal professor da IGTI da época eu levantei todos os tópicos que para mim na época faziam ser de extrema necessidade para quando você fala que pretende fazer uma pós graduação em Front-end, veja se faz sentido para que tu:
```
- ES6/ES7
- JavasScript Isomorfico, 
- Virtual Dom
- Webpack, Parcel e Gulp 
- Padrões de projeto
- Arquitetura DDD e Hexagonal
- Clean Code 
- SOLID
- KISS
- TDD

Ferramentas para performance e otimização para client-side, prevenir e garantir uma qualidade e segurança da aplicação para evitar XSS.

- ITCSS
- BEM 
- SMACSS
- OOCSS
- FlexBox
- CSSGrid
- Atomic Design

 E acredito que por conta do mercado também é necessário estar engajado em 1 ou 2 frameworks de maior adesão no caso e conhecendo mais sobre alguma framework de back-end como meu interesse maior é por front-end, acredito que conhecer Node seria super útil além de:

- React/ Redux
- Vue/ Vuex. 
- Jasmine / Jest ( para testes unitários) 

Também outros pontos que acho relevante para ser abordado seria o contexto de:
- PWA
- JWT
- Websocket
``` 

É isso.

**Comment by vmagalhaes** [comment]
@onlurking @cristofersousa valeu pelas dicas, depois de algum estudo e do comentário de vocês desisti de fazer alguma pós relacionada a Dev mesmo, vou focar nos cursos de Front e Dev, vou continuar me aprofundando em relação ao Desenvolvimento Javascript, estudar mais sobre Node, Vue e etc. Hoje eu tenho um conhecimento focado à Angular, mas o mercado espera também sobre React e Vue, acredito que me profissionalizar nesses frameworks vai me abrir um caminho bem legal.

Valeu!

**Comment by pierryangelo** [comment]
> Cara, depende muito do teu foco, se é se especializar em HTML, Acessibilidade, CSS ou Javascript(vanilla ou em algum framework em específico por exemplo), tem que escolher um e se aprofundar no melhores materiais disponíveis sobre o assunto, é impossível esperar que uma pós ou bootcamp cubra esses assuntos decentemente.
> 
> De CSS por exemplo tem o [The CSS Layout Workshop](https://thecssworkshop.com/toc) da [Rachel Andrew](https://rachelandrew.co.uk/css/) que é uma das mentes por trás do CSS Grid Layout e ela está no CSS Working Group, que é o comitê responsável por criar as especificações do CSS.
> 
> Tem o [The Responsive Web Design Bootcamp](https://scrimba.com/g/gresponsive) do [Kevin Powell](https://www.youtube.com/channel/UCJZv4d5rbIKd4QHMPkcABCw), que é uma referência de CSS também, recomendo dar uma olhada no canal dele no youtube.
> 
> De referência nacional em CSS os cursos do [André Rafael da Origamid](https://www.origamid.com/) são incríveis, tenho o certificado de Flexbox e Grid Layout dele, e só pra tu ter uma ideia, botou o nanodegree de front-end que fiz da Udacity(material feito em parceria pela Google) no chinelo, se quiser ter uma noção da qualidade dá uma olhada nos guias de referência dele em [flexbox](https://origamid.com/projetos/flexbox-guia-completo/) e [grid layout](https://www.origamid.com/projetos/css-grid-layout-guia-completo/).
> 
> De Javascript tem os materiais do [Eric Elliot](https://ericelliottjs.com), que é o cara por trás de vários artigos incríveis de Javascript como o [The Two Pillars of JavaScript](https://medium.com/javascript-scene/the-two-pillars-of-javascript-ee6f3281e7f3), tem os livros da série [You Don't Know Javascript](https://github.com/getify/You-Dont-Know-JS#titles) do Kyle Simpson e também os cursos dele no [Frontend Masters](https://frontendmasters.com/teachers/kyle-simpson/), conteúdo nacional tem os cursos do [Daciuk](https://queroser.ninja/promocoes/) e do [Willian Justen](https://www.udemy.com/user/willian-justen-de-vasconcellos/) que são duas referências em frontend.
> 
> De React tem os cursos do [Tyler Mcginnis](https://tylermcginnis.com/courses/) que está por trás da emenda do [nanodegree de React](https://www.udacity.com/course/react-nanodegree--nd019) da Udacity, de conteúdo nacional de React tem o curso [React Ninja](https://www.udemy.com/course/curso-reactjs-ninja/) do Fernando Daciuk, os cursos do [devPleno](https://www.devpleno.com) do Tulio Faria, tem o material da [Rocketseat](https://rocketseat.com.br)(estou fazendo o bootcamp deles atualmente, é bem legal mas peca bastante em boas práticas).
> 
> Certificados oficiais em frontend não existem, quer dizer, tem os certificados da [W3C pela edX](https://www.edx.org/professional-certificate/front-end-web-developer-9) mas os cursos são bem básicos, não valem a pena, e tem os da [W3Schools](https://www.w3schools.com/cert/) mas eles tem a imagem um pouco suja por ter um histórico de [ensinar coisas erradas](https://web.archive.org/web/20120502220433/http://w3fools.com/).
> 
> De qualquer maneira, portfolio é muito mais relevante que qualquer certificado. Vai lhe agregar muito mais você escolher um curso/bootcamp que é recheado de projetos, fazer eles, colocar no Github, e melhorar eles após a finalização do curso(pesquisar sobre boas práticas, documentação e refatorar, procurar material fora do curso, na minha opinião o aprendizado real acontece aqui).

Excelentes dicas, @onlurking!

**Comment by vmagalhaes** [comment]
5 anos depois posso dizer, a pós graduação não teria afetado na minha carreira. Obviamente depende de pessoa para pessoa, mas no geral, estude sobre o que quer focar, a melhoria e crescimento é possível com as oportunidades que a sua própria carreira vai te dar. Hoje continuo trabalhando com Angular, com uma grande expertise, o que me fez estar hoje posso estar, mas claro, abra os horizontes de conhecimento, é muito importante.
